### PR TITLE
Add refactored docker auth plugin implementation

### PR DESCRIPTION
**Pull request by tharindulak**
OS: Mac OS 10.14.5

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# ------------------------------------------------------------------------
```
Comment (nadundesilva) [inline Dockerfile]: Why do we need this Dockerfile at the root ?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+#
+# ------------------------------------------------------------------------
+
+FROM wso2cellery/auth-server:latest AS build-env
+COPY ./components/docker-auth /go/src/github.com/cellery-io/cellery-hub/components/docker-auth/
+RUN cd /go/src/github.com/cellery-io/cellery-hub/components/docker-auth/ && echo "replace github.com/cesanta/docker_auth/auth_server v0.0.0-20190831165929-82573a5f102c => /go/src/github.com/cesanta/docker_auth/auth_server" >> go.mod
```
Comment (nadundesilva) [inline Dockerfile]: Why do we need to update the mod file ?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+RUN export GO111MODULE=on && cd /go/src/github.com/cellery-io/cellery-hub/components/docker-auth/ && go build -buildmode=plugin -o /plugins/authz.so cmd/authz/authorization.go
+RUN export GO111MODULE=on && cd /go/src/github.com/cellery-io/cellery-hub/components/docker-auth/ && go build -buildmode=plugin -o /plugins/authn.so cmd/authn/authentication.go
+
+FROM ubuntu:18.04
+COPY --from=build-env /go/src/github.com/cesanta/docker_auth/auth_server/main /
+COPY --from=build-env /plugins/ /plugins/
+
+ENTRYPOINT ["/main"]
+CMD ["/config/auth_config.yml"]
```
Comment (nadundesilva) [inline Dockerfile]: Why is this required ?

Default command pointing to a yaml doesn't make sense, IINM.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License
+#
+# ------------------------------------------------------------------------
+
+FROM wso2cellery/auth-server:latest AS build-env
+COPY ./components/docker-auth /go/src/github.com/cellery-io/cellery-hub/components/docker-auth/
+RUN cd /go/src/github.com/cellery-io/cellery-hub/components/docker-auth/ && echo "replace github.com/cesanta/docker_auth/auth_server v0.0.0-20190831165929-82573a5f102c => /go/src/github.com/cesanta/docker_auth/auth_server" >> go.mod
+RUN export GO111MODULE=on && cd /go/src/github.com/cellery-io/cellery-hub/components/docker-auth/ && go build -buildmode=plugin -o /plugins/authz.so cmd/authz/authorization.go
+RUN export GO111MODULE=on && cd /go/src/github.com/cellery-io/cellery-hub/components/docker-auth/ && go build -buildmode=plugin -o /plugins/authn.so cmd/authn/authentication.go
```
Comment (nadundesilva) [inline Dockerfile]: You can set the env variable once.

Also can set the working directory to /go/src/github.com/cellery-io/cellery-hub/components/docker-auth/ at the start.

This will make the instructions much more clearer.

File: Makefile [hunk 5]
```diff
@@ -210,18 +208,10 @@ docker.$(PROXY): build.$(PROXY)
...
-	rm -rf ./docker/$(DOCKER_AUTH)/$(AUTH_SERVER)/target
-	mkdir ./docker/$(DOCKER_AUTH)/target/
-	cp ./components/$(DOCKER_AUTH)/target/authentication ./docker/$(DOCKER_AUTH)/target/
-	cp ./components/$(DOCKER_AUTH)/target/authorization ./docker/$(DOCKER_AUTH)/target/
-	mkdir ./docker/$(DOCKER_AUTH)/$(AUTH_SERVER)/target/
-	cp ./components/$(DOCKER_AUTH)/target/auth_server ./docker/$(DOCKER_AUTH)/$(AUTH_SERVER)/target/
-	cd ./docker/$(DOCKER_AUTH); \
+docker.$(DOCKER_AUTH):
+	cd ./docker/$(AUTH_SERVER); \
+	docker build -t $(DOCKER_REPO)/$(AUTH_SERVER):latest .
```
Comment (nadundesilva) [inline Makefile]: Don't hard-code the tag. Please use the VERSION variable.

File: Dockerfile (hunk 1 above)
Comment (tharindulak) [reply Dockerfile]: Bought the file to the docker-auth folder

File: Dockerfile (hunk 2 above)
Comment (tharindulak) [reply Dockerfile]: We need to point our docker-auth code. otherwise, code will not match

File: Dockerfile (hunk 3 above)
Comment (tharindulak) [reply Dockerfile]: Removed this one

File: Dockerfile (hunk 4 above)
Comment (tharindulak) [reply Dockerfile]: Fixed

File: Makefile (hunk 5 above)
Comment (tharindulak) [reply Makefile]: Used the commit hash as the image version. Since we need to keep this base image separate.

File: components/docker-auth/cmd/authz/authorization.go [hunk 6]
```diff
@@ -19,88 +19,61 @@
...
+	if err != nil {
+		return nil, fmt.Errorf("error in generating the execId : %s", err)
 	}
-	if res.StatusCode == http.StatusOK {
-		log.Printf("[%s] Authorized request. Exiting with success exit code", execId)
-		os.Exit(extension.SuccessExitCode)
+	logger.Debugf("Authorization logic reached. User will be authorized")
+	dbConnectionPool, err := db.GetDbConnectionPool(logger)
+	if err != nil {
+		return nil, fmt.Errorf("error while establishing database connection pool")
```
Comment (Mirage20) [inline components/docker-auth/cmd/authz/authorization.go]: Add the original err when returning new error with messages

File: components/docker-auth/cmd/authz/authorization.go [hunk 7]
```diff
@@ -19,88 +19,61 @@
...
-// resolves the authorization end point from the environment variables.
-func resolveAuthorizationUrl(execId string) string {
-	authServer := os.Getenv("AUTH_SERVER_URL")
-	authorizationEP := os.Getenv("AUTHORIZATION_END_POINT")
-	if len(authServer) == 0 {
-		log.Printf("[%s] Error: AUTH_SERVER environment variable is empty\n", execId)
-		return ""
+	authorized, err := auth.Authorization(dbConnectionPool, ai, logger, execId)
+	if err != nil {
+		return nil, fmt.Errorf("error while executing authorization logic")
```
Comment (Mirage20) [inline components/docker-auth/cmd/authz/authorization.go]: Same

File: components/docker-auth/pkg/auth/authn_handler.go [hunk 8]
```diff
@@ -42,160 +40,168 @@ const (
...
-func isJWT(execId string) bool {
+func isJWT(logger *zap.SugaredLogger, execId string) (bool, error) {
 	isJWTEnv := os.Getenv("IS_JWT")
 	var isJWT bool
 	if len(isJWTEnv) == 0 {
-		log.Printf("[%s] Error: IS_JWT environment variable is empty\n", execId)
-		return false
+		return false, fmt.Errorf("error: IS_JWT environment variable is empty")
 	} else {
 		if isJWTEnv == "true" {
```
Comment (Mirage20) [inline components/docker-auth/pkg/auth/authn_handler.go]: Can't we use https://golang.org/pkg/strconv/#ParseBool to parse the boolean?

File: components/docker-auth/pkg/auth/authn_handler.go [hunk 9]
```diff
@@ -42,160 +40,168 @@ const (
...
 }
 
 // validateAccessToken is used to introspect the access token
-func validateAccessToken(token string, providedUsername string, execId string) bool {
-	introspectionUrl, urlAvailability := resolveIntrospectionUrl(execId)
+func validateAccessToken(token string, providedUsername string, logger *zap.SugaredLogger,
+	execId string) (bool, error) {
+	introspectionUrl, urlAvailability, err := resolveIntrospectionUrl(logger, execId)
+	if err != nil {
+		return false, fmt.Errorf("something went wrong while resolving introspection url")
```
Comment (Mirage20) [inline components/docker-auth/pkg/auth/authn_handler.go]: Missing original err

File: components/docker-auth/pkg/auth/authn_handler.go [hunk 10]
```diff
@@ -207,112 +213,108 @@ func validateAccessToken(token string, providedUsername string, execId string) b
...
+		return false, fmt.Errorf("error un marshalling the json : %s", err)
 	}
 	isActive, ok := (result["active"]).(bool)
 	if !ok {
-		log.Printf("[%s] Error casting active to boolean. This may be due to a invalid token\n", execId)
-		return false
+		return false, fmt.Errorf("error casting active to boolean. This may be due to a invalid token")
+	}
+	logger.Debugf("[%s] Resolved acess token validity", execId)
+	isExpired, err := isExpired(result["exp"], logger, execId)
```
Comment (Mirage20) [inline components/docker-auth/pkg/auth/authn_handler.go]: Is there any reason to use map[string]interface{} without using struct?

File: Makefile [hunk 11]
```diff
@@ -246,10 +249,13 @@ docker.$(IDENTITY_SERVER_CUSTOMIZATION): build.$(IDENTITY_SERVER_CUSTOMIZATION)
 docker-push.$(PROXY):
 	docker push $(DOCKER_REPO)/cellery-hub-proxy:$(VERSION)
 
+.PHONY: docker-push.$(DOCKER_AUTH_BASE)
+docker-push.$(DOCKER_AUTH_BASE):
+	docker push $(DOCKER_REPO)/$(DOCKER_AUTH_BASE):82573a5f102c2d886428415bf6fcb8d94706834a
```
[the remark this answers is not in the document]
Comment (tharindulak) [reply Makefile]: Fixed

File: components/docker-auth/cmd/authn/authentication.go [hunk 12]
```diff
@@ -19,123 +19,84 @@
...
+	"go.uber.org/zap"
+
+	"github.com/cellery-io/cellery-hub/components/docker-auth/pkg/auth"
 	"github.com/cellery-io/cellery-hub/components/docker-auth/pkg/extension"
 )
 
-const (
-	logFile = "/extension-logs/authn-ext.log"
-)
+var authnLogger *zap.SugaredLogger
```
[the remark this answers is not in the document]
Comment (tharindulak) [reply components/docker-auth/cmd/authn/authentication.go]: Fixed

File: components/docker-auth/cmd/authn/authentication.go [hunk 13]
```diff
@@ -19,123 +19,84 @@
...
+
+type PluginAuthn struct {
+}
 
-func main() {
-	err := os.MkdirAll("/extension-logs", os.ModePerm)
-	file, err := os.OpenFile(logFile, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
-	defer func() {
-		err = file.Close()
+func (c *PluginAuthn) Authenticate(user string, password api.PasswordString) (bool, api.Labels, error) {
```
[the remark this answers is not in the document]
Comment (tharindulak) [reply components/docker-auth/cmd/authn/authentication.go]: Fixed

File: components/docker-auth/cmd/authn/authentication.go [hunk 14]
```diff
@@ -19,123 +19,84 @@
...
+	if authnLogger == nil {
+		var err error
+		authnLogger, err = extension.NewLogger()
 		if err != nil {
-			log.Printf("Error while closing the file : %s\n", err)
-			os.Exit(extension.MisuseExitCode)
+			log.Fatalf("can't initialize zap logger: %v", err)
```
[the remark this answers is not in the document]
Comment (tharindulak) [reply components/docker-auth/cmd/authn/authentication.go]: Fixed

File: components/docker-auth/cmd/authn/authentication.go [hunk 15]
```diff
@@ -19,123 +19,84 @@
...
 	if err != nil {
-		log.Printf("Error in generating the execId : %s\n", err)
-		os.Exit(extension.ErrorExitCode)
+		return false, nil, fmt.Errorf("error in generating the execId : %s", err)
 	}
 
-	text := extension.ReadStdIn()
-	log.Printf("[%s] Payload received from CLI : %s\n", execId, text)
-	credentials := strings.Split(text, " ")
+	logger.Debugf("[%s] Username (%s) and password received from CLI", execId, user)
```
[the remark this answers is not in the document]
Comment (tharindulak) [reply components/docker-auth/cmd/authn/authentication.go]: Fixed

File: components/docker-auth/cmd/authn/authentication.go [hunk 16]
```diff
@@ -19,123 +19,84 @@
...
+		logger.Debugf("[%s] Ping request received", execId)
 	}
 
-	url := resolveAuthenticationUrl(execId)
-	if url == "" {
-		log.Printf("[%s] Authentication end point not found. Exiting with error exit code", execId)
-		os.Exit(extension.ErrorExitCode)
+	isAuthenticated, err := auth.Authenticate(user, token, logger, execId)
+	if err != nil {
+		return false, nil, fmt.Errorf("something went wrong while authenticating %v", err)
```
[the remark this answers is not in the document]
Comment (tharindulak) [reply components/docker-auth/cmd/authn/authentication.go]: Fixed

File: components/docker-auth/cmd/authn/authentication.go [hunk 17]
```diff
@@ -19,123 +19,84 @@
...
-				" authorization filter\n", execId)
-			os.Exit(extension.ErrorExitCode)
+			return false, nil, fmt.Errorf("since this is a ping request, exiting with auth fail status " +
+				"without passing to authorization filter")
 		} else {
-			log.Printf("[%s] Failed authentication. But passing to authorization filter", execId)
-			addAuthenticationLabel(false, execId)
-			os.Exit(extension.SuccessExitCode)
+			logger.Debugf("[%s] Failed authentication. But passing to authorization filter", execId)
+			return true, addAuthenticationLabel(false), nil
```
[the remark this answers is not in the document]
Comment (tharindulak) [reply components/docker-auth/cmd/authn/authentication.go]: Fixed

File: components/docker-auth/cmd/authn/authentication.go [hunk 18]
```diff
@@ -19,123 +19,84 @@
...
 		}
-	}
-	if res.StatusCode == http.StatusOK {
-		log.Printf("[%s] User successfully authenticated by validating token. Exiting with success exit code",
-			execId)
-		addAuthenticationLabel(true, execId)
-		os.Exit(extension.SuccessExitCode)
+	} else {
+		logger.Debugf("[%s] User successfully authenticated by validating token. Exiting with success "+
```
[the remark this answers is not in the document]
Comment (tharindulak) [reply components/docker-auth/cmd/authn/authentication.go]: Fixed

File: components/docker-auth/cmd/authn/authentication.go [hunk 19]
```diff
@@ -19,123 +19,84 @@
...
-	log.Printf("[%s] Adding labels to authorization ext from authn ext: %s\n", execId, label)
-	_, err := os.Stdout.WriteString(label)
-	if err != nil {
-		log.Printf("[%s] Error in writing to standard output. Hence failing authentication. No authorizatino done", err)
-		os.Exit(extension.ErrorExitCode)
-	}
+	label := make([]string, 1)
+	label[0] = authResultString
+	authLabels := api.Labels{}
+	authLabels["isAuthSuccess"] = label
```
[the remark this answers is not in the document]
Comment (tharindulak) [reply components/docker-auth/cmd/authn/authentication.go]: Fixed

File: components/docker-auth/cmd/authz/authorization.go [hunk 20]
```diff
@@ -19,88 +19,61 @@
...
-
-	execId, err := extension.GetExecID()
-	if err != nil {
-		log.Printf("Error in generating the execId : %s\n", err)
-		os.Exit(extension.ErrorExitCode)
-	}
-
-	accessToken := extension.ReadStdIn()
-	log.Printf("[%s] Access token received\n", execId)
+var authzLogger *zap.SugaredLogger
```
[the remark this answers is not in the document]
Comment (tharindulak) [reply components/docker-auth/cmd/authz/authorization.go]: Fixed

File: components/docker-auth/cmd/authz/authorization.go (hunk 6 above)
Comment (tharindulak) [reply components/docker-auth/cmd/authz/authorization.go]: Fixed

File: components/docker-auth/cmd/authz/authorization.go (hunk 7 above)
Comment (tharindulak) [reply components/docker-auth/cmd/authz/authorization.go]: Fixed

File: components/docker-auth/pkg/auth/authn_handler.go (hunk 8 above)
Comment (tharindulak) [reply components/docker-auth/pkg/auth/authn_handler.go]: Fixed

File: components/docker-auth/pkg/auth/authn_handler.go (hunk 9 above)
Comment (tharindulak) [reply components/docker-auth/pkg/auth/authn_handler.go]: Fixed

File: components/docker-auth/pkg/auth/authn_handler.go (hunk 10 above)
Comment (tharindulak) [reply components/docker-auth/pkg/auth/authn_handler.go]: Fixed

File: components/docker-auth/pkg/auth/authn_handler.go [hunk 21]
```diff
@@ -42,160 +41,164 @@ const (
...
-		log.Printf("[%s] Authentication failed\n", execId)
-		return false
+		logger.Debugf("[%s] Authentication failed", execId)
+		return false, nil
 	}
 }
 
 // isJWT checks whether the token is jwt token or access token.
-func isJWT(execId string) bool {
+func isJWT() (bool, error) {
```
Comment (hasinthaindrajee) [inline components/docker-auth/pkg/auth/authn_handler.go]: This function doesn't have a meaningful name. This is supposed to check whether we need to validate JWT as well. We can remove this as well since this is no longer used.

File: components/docker-auth/pkg/auth/authn_handler.go [hunk 22]
```diff
@@ -207,112 +210,105 @@ func validateAccessToken(token string, providedUsername string, execId string) b
...
+	if err != nil {
+		return false, err
+	}
+	return isExpired && response.Active && isValidUser, nil
 }
 
 // resolves the IS host and port from the environment variables.
 // If the environment is not set the port and host will be resolved through the config file.
-func resolveIntrospectionUrl(execId string) (string, bool) {
+func resolveIntrospectionUrl(logger *zap.SugaredLogger, execId string) (string, bool, error) {
```
Comment (hasinthaindrajee) [inline components/docker-auth/pkg/auth/authn_handler.go]: Why we need 3 things returned from here ? Returning URL would be enough. (you may have an error as well)

File: components/docker-auth/pkg/auth/authn_handler.go [hunk 23]
```diff
@@ -42,160 +41,164 @@ const (
...
 	if err != nil {
-		log.Printf("[%s] Error creating new request to the introspection endpoint : %s\n", execId, err)
-		return false
+		return false, fmt.Errorf("error creating new request to the introspection endpoint : %s", err)
 	}
-	log.Printf("[%s] Created new request to the introspection endpoint : %s\n", execId, introspectionUrl)
-	username, password, credentialsAvailability := resolveCredentials(execId)
-	if !credentialsAvailability {
-		return false
+	logger.Debugf("[%s] Created new request to the introspection endpoint : %s", execId, introspectionUrl)
```
Comment (hasinthaindrajee) [inline components/docker-auth/pkg/auth/authn_handler.go]: This log looks redundant.

File: components/docker-auth/pkg/auth/authn_handler.go [hunk 24]
```diff
@@ -207,112 +210,105 @@ func validateAccessToken(token string, providedUsername string, execId string) b
...
 	}
-	return idpEndPoint + introspectionEP, true
+	logger.Debugf("[%s] Successfully resolved introspection url", execId)
+	return idpEndPoint + introspectionEP, true, nil
 }
 
 // resolveCredentials resolves the user credentials of the user that is used to communicate to introspection endpoint
-func resolveCredentials(execId string) (string, string, bool) {
+func resolveCredentials(logger *zap.SugaredLogger, execId string) (string, string, error) {
 	username := os.Getenv("USERNAME")
```
Comment (hasinthaindrajee) [inline components/docker-auth/pkg/auth/authn_handler.go]: Can't we define them as constants ?

File: components/docker-auth/pkg/auth/authn_handler.go [hunk 25]
```diff
@@ -207,112 +210,105 @@ func validateAccessToken(token string, providedUsername string, execId string) b
...
-		return "", "", true
+		return "", "", fmt.Errorf("'USERNAME' environment variable is empty")
 	}
 	password := os.Getenv("PASSWORD")
 	if len(password) == 0 {
-		log.Printf("[%s] Error: PASSWORD environment variable is empty\n", execId)
-		return "", "", true
+		return "", "", fmt.Errorf("'PASSWORD' environment variable is empty")
 	}
-	log.Printf("[%s] Suceesfully received credentials\n", execId)
```
Comment (hasinthaindrajee) [inline components/docker-auth/pkg/auth/authn_handler.go]: does **received** make sense ?

File: components/docker-auth/pkg/auth/authz_handler.go [hunk 26]
```diff
@@ -19,24 +19,27 @@
...
+	"github.com/cesanta/docker_auth/auth_server/api"
+	"go.uber.org/zap"
 
 	"github.com/cellery-io/cellery-hub/components/docker-auth/pkg/extension"
 )
 
-func Authorization(dbConn *sql.DB, accessToken string, execId string) int {
-	log.Printf("[%s] Authorization logic handler reached and access will be validated\n", execId)
-	isValid, err := extension.ValidateAccess(dbConn, accessToken, execId)
+func Authorization(dbConn *sql.DB, ai *api.AuthRequestInfo, logger *zap.SugaredLogger, execId string) (bool, error) {
```
Comment (hasinthaindrajee) [inline components/docker-auth/pkg/auth/authz_handler.go]: Is this function coming from an interface ? If this is defined by you can we change the name of the function >

File: components/docker-auth/pkg/extension/acl.go [hunk 27]
```diff
@@ -20,239 +20,229 @@ package extension
...
 	"strings"
 
+	"go.uber.org/zap"
+
+	"github.com/cesanta/docker_auth/auth_server/api"
 	_ "github.com/go-sql-driver/mysql"
 )
 
-type Labelstest map[string][]string
+func ValidateAccess(db *sql.DB, actions []string, username string, repository string, labels api.Labels,
```
Comment (hasinthaindrajee) [inline components/docker-auth/pkg/extension/acl.go]: Shall we change this to isUserAuthorized ? Always try to use meaningful short names for methods

File: components/docker-auth/pkg/extension/acl_test.go [hunk 28]
```diff
@@ -47,7 +52,7 @@ func createConn() bool {
 	dbConnection, err = sql.Open(dbDriver, dbUser+":"+dbPass+"@tcp("+host+":"+port+")/"+dbName)
 	if err != nil {
 		fmt.Println("Error while connecting to the database")
-		os.Exit(ErrorExitCode)
+		os.Exit(1)
```
Comment (hasinthaindrajee) [inline components/docker-auth/pkg/extension/acl_test.go]: I think the previous way is more readable. When an os.Exist happens a person might not know exactly what 1 means

File: components/docker-auth/pkg/extension/acl.go [hunk 29]
```diff
@@ -20,239 +20,229 @@ package extension
...
+	logger.Debugf("[%s] User %s is trying to perform delete action on organization :%s", execId, user,
 		organization)
 	results, err := db.Query(getUserRoleQuery, user, organization)
 	defer func() {
-		closeResultSet(results, "isAuthorizedToDelete", execId)
+		closeResultSet(results, "isAuthorizedToDelete", logger, execId)
 	}()
 	if err != nil {
-		log.Printf("[%s] Error while calling the mysql query for getting the user role :%s\n", execId, err)
+		logger.Debugf("[%s] Error while calling the mysql query for getting the user role :%s", execId, err)
```
Comment (hasinthaindrajee) [inline components/docker-auth/pkg/extension/acl.go]: Error while executing ?

File: components/docker-auth/pkg/extension/acl.go [hunk 30]
```diff
@@ -20,239 +20,229 @@ package extension
...
+	logger.Debugf("[%s] User %s is trying to push to organization :%s", execId, user, organization)
 	results, err := db.Query(getUserRoleQuery, user, organization)
 	defer func() {
-		closeResultSet(results, "isAuthorizedToPush", execId)
+		closeResultSet(results, "isAuthorizedToPush", logger, execId)
 	}()
 	if err != nil {
-		log.Printf("[%s] Error while calling the mysql query getUserRoleQuery :%s\n", execId, err)
-		return false, err
+		return false, fmt.Errorf("[%s] Error while calling the mysql query getUserRoleQuery :%s", execId, err)
```
Comment (hasinthaindrajee) [inline components/docker-auth/pkg/extension/acl.go]: calling ? or executing ?

File: components/docker-auth/pkg/extension/acl.go [hunk 31]
```diff
@@ -20,239 +20,229 @@ package extension
...
 
-	visibility, err := getImageVisibility(db, image, organization, execId)
+	visibility, err := getImageVisibility(db, image, organization, logger, execId)
 
 	if err != nil {
-		log.Printf("[%s] User %s is not authorized to pull the image %s/%s.\n", execId, user,
+		logger.Debugf("[%s] User %s is not authorized to pull the image %s/%s.", execId, user,
 			organization, image)
-		return false, fmt.Errorf("error occured while geting visibility of image %s/%s", organization, image)
+		return false, fmt.Errorf("error occured while geting visibility of image. User %s is not "+
```
Comment (hasinthaindrajee) [inline components/docker-auth/pkg/extension/acl.go]: Shall we use one convention ? Either start with capital or sample

File: components/docker-auth/pkg/auth/authn_handler.go (hunk 21 above)
Comment (tharindulak) [reply components/docker-auth/pkg/auth/authn_handler.go]: Fixed by #159

File: components/docker-auth/pkg/auth/authn_handler.go (hunk 22 above)
Comment (tharindulak) [reply components/docker-auth/pkg/auth/authn_handler.go]: Fixed by #159

File: components/docker-auth/pkg/auth/authn_handler.go (hunk 23 above)
Comment (tharindulak) [reply components/docker-auth/pkg/auth/authn_handler.go]: Fixed by #159

File: components/docker-auth/pkg/auth/authn_handler.go (hunk 24 above)
Comment (tharindulak) [reply components/docker-auth/pkg/auth/authn_handler.go]: Fixed by #159

File: components/docker-auth/pkg/auth/authn_handler.go (hunk 25 above)
Comment (tharindulak) [reply components/docker-auth/pkg/auth/authn_handler.go]: Fixed by #159

File: components/docker-auth/pkg/auth/authz_handler.go (hunk 26 above)
Comment (tharindulak) [reply components/docker-auth/pkg/auth/authz_handler.go]: Fixed by #159

File: components/docker-auth/pkg/extension/acl.go (hunk 27 above)
Comment (tharindulak) [reply components/docker-auth/pkg/extension/acl.go]: Fixed by #159

File: components/docker-auth/pkg/extension/acl_test.go (hunk 28 above)
Comment (tharindulak) [reply components/docker-auth/pkg/extension/acl_test.go]: Fixed by #159

File: components/docker-auth/pkg/extension/acl.go (hunk 29 above)
Comment (tharindulak) [reply components/docker-auth/pkg/extension/acl.go]: Fixed by #159

File: components/docker-auth/pkg/extension/acl.go (hunk 30 above)
Comment (tharindulak) [reply components/docker-auth/pkg/extension/acl.go]: Fixed by #159

File: components/docker-auth/pkg/extension/acl.go (hunk 31 above)
Comment (tharindulak) [reply components/docker-auth/pkg/extension/acl.go]: Fixed by #159
